### PR TITLE
Update Observatory to provide detailed stats about grade improvements

### PR DESCRIPTION
**Pull request by april**

### CODE REVIEW

File: httpobs/database/database.py [hunk 1]
```diff
@@ -180,10 +184,10 @@ def periodic_maintenance() -> int:
         return cur.rowcount
 
 
-def select_scan_grade_totals() -> dict:
-    # Used for /api/v1/getGradeDistribution
+def select_star_from(table: str) -> dict:
+    # Select all the rows in a given table. Note that this is specifically not parameterized.
```
Comment (floatingatoll) [inline httpobs/database/database.py]: Why not? (No objections, but maybe the next line explaining went missing / doesn't exist?)

File: httpobs/database/database.py [hunk 2]
```diff
@@ -210,11 +214,25 @@ def select_scan_host_history(site_id: int) -> list:
...
+        states = dict(cur.fetchall())
+
+        # Get the recent scan count
+        cur.execute("""SELECT DATE_TRUNC('hour', end_time) AS hour, COUNT(*) as num_scans
+                         FROM scans
+                         WHERE (end_time < DATE_TRUNC('hour', NOW()))
+                           AND (end_time >= DATE_TRUNC('hour', NOW()) - INTERVAL '24 hours')
+                         GROUP BY hour
+                         ORDER BY hour DESC;""",
+                    (STATE_FINISHED,))
```
Comment (floatingatoll) [inline httpobs/database/database.py]: Is the trailing comma here intentional / part of this syntax commonly?

File: httpobs/database/schema.sql [hunk 3]
```diff
@@ -84,6 +84,7 @@ CREATE MATERIALIZED VIEW latest_scans
   LATERAL ( SELECT id AS scan_id, site_id, state, start_time, end_time, tests_failed, tests_passed, grade, score, error
             FROM scans WHERE site_id = s.id AND state = 'FINISHED' ORDER BY end_time DESC LIMIT 1 ) latest_scans;
 COMMENT ON MATERIALIZED VIEW latest_scans IS 'Most recently completed scan for a given website';
+GRANT SELECT ON latest_scans TO httpobsapi;
```
Comment (floatingatoll) [inline httpobs/database/schema.sql]: I couldn't find the change above that would imply this SELECT grant was necessary. Does this need to be done for all materialized views, or just this one, or ..?

File: httpobs/database/schema.sql [hunk 4]
```diff
@@ -111,3 +111,43 @@ GRANT UPDATE (cookies) ON sites TO httpobsapi;
 /*
 ALTER TABLE scans ADD COLUMN likelihood_indicator VARCHAR NULL;
 */
+
+/* Update to frequently refresh latest_scans */
+/*
+GRANT SELECT ON latest_scans TO httpobsapi;
```
Comment (floatingatoll) [inline httpobs/database/schema.sql]: Is this a commented-out dupe of the above?

File: httpobs/database/database.py (hunk 1 above)
Comment (april) [inline httpobs/database/database.py]: You can't parameterize the table name.  :)

File: httpobs/database/database.py (hunk 2 above)
Comment (april) [inline httpobs/database/database.py]: It only accepts a tuple:

```
>>> a = (5)
>>> type(a)
<type 'int'>
>>> a = (5,)
>>> type(a)
<type 'tuple'>
```

File: httpobs/database/schema.sql (hunk 3 above)
Comment (april) [inline httpobs/database/schema.sql]: Just trying to be consistent with granting read access to these materialized views, as they only contain useful statistics.

File: httpobs/database/schema.sql (hunk 4 above)
Comment (april) [inline httpobs/database/schema.sql]: Nope, just a reminder that this was done to a live system. I'll probably go through and remove these eventually.

**Pull request merged**